cmd/virt-launcher/virt-launcher.go

```diff
@@ -367,6 +386,7 @@ func main() {
 
 	// Initialize local and shared directories
 	initializeDirs(*virtShareDir, *ephemeralDiskDir, *uid)
+	defer cleanupEphemeralDiskDirectory(*ephemeralDiskDir)
```

Comment:
How would you feel about placing this cleanup in the "ForkAndMonitor()" function?

virt-launcher is using a pattern where the first execution forks and monitors a second virt-launcher process. This lets us intercept things like SIG_TERM for graceful shutdown, and properly handle/report virt-launcher crashes.

For this to work, some minor refactoring to ForkAndMonitor would need to occur. Rather than having FortAndMonitor call os.Exit() directly, we'd need ForkAndMonitor to return an exit code and an error and let Main() call os.Exit().  This would let your "defer" statement work as expected.

Reply:
Ok, sure. I can move it there.
My only concern is that we won't clean up on panic.
This is why I've placed it here in main(), so no matter what, we will do the clean up before exiting.